x-pack/agent/pkg/fleetapi/checkin_cmd_test.go

```diff
@@ -0,0 +1,266 @@
+				w.WriteHeader(http.StatusOK)
+
+				type E struct {
+					ActionID  string    `json:"action_id"`
+					Type      string    `json:"type"`
+					Message   string    `json:"message"`
+					Timestamp time.Time `json:"timestamp"`
+				}
+
+				responses := struct {
+					Events []E `json:"events"`
+				}{}
+
+				decoder := json.NewDecoder(r.Body)
+				defer r.Body.Close()
+
+				err := decoder.Decode(&responses)
+				require.NoError(t, err)
+
+				require.Equal(t, 1, len(responses.Events))
+
+				e := responses.Events[0]
+				require.Equal(t, "my-id", e.ActionID)
+				require.Equal(t, "ACTION_ACKNOWLEDGED", e.Type)
+				require.Equal(t, "Acknowledge action my-id", e.Message)
+
+				fmt.Fprintf(w, raw)
+			}, withAPIKey))
+			return mux
+		}, withAPIKey,
+		func(t *testing.T, client clienter) {
+			action := &PolicyChangeAction{
+				BaseAction: &BaseAction{
+					ActionID:   "my-id",
+					ActionType: "POLICY_CHANGE",
+				},
+				Policy: map[string]interface{}{
+					"id": "policy_id",
+				},
+			}
+
+			cmd := NewCheckinCmd(agentID, client)
+
+			request := CheckinRequest{
+				Events: []SerializableEvent{
+					Ack(action),
+				},
+			}
+
+			r, err := cmd.Execute(&request)
+			require.NoError(t, err)
+			require.True(t, r.Success)
+
+			require.Equal(t, 0, len(r.Actions))
+		},
+	))
+
+	t.Run("Propagate any errors from the server", withServerWithAuthClient(
+		func(t *testing.T) *http.ServeMux {
+			raw := `
+Something went wrong
+}
+`
+			mux := http.NewServeMux()
+			path := fmt.Sprintf("/api/fleet/agents/%s/checkin", agentID)
+			mux.HandleFunc(path, authHandler(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusInternalServerError)
+				fmt.Fprintf(w, raw)
+			}, withAPIKey))
+			return mux
+		}, withAPIKey,
+		func(t *testing.T, client clienter) {
+			cmd := NewCheckinCmd(agentID, client)
+
+			request := CheckinRequest{}
+
+			_, err := cmd.Execute(&request)
+			require.Error(t, err)
+		},
+	))
+
+	t.Run("Checkin receives a PolicyChange", withServerWithAuthClient(
+		func(t *testing.T) *http.ServeMux {
+			raw := `
+{
+    "actions": [
+        {
+            "type": "POLICY_CHANGE",
+            "id": "id1",
+            "data": {
+                "policy": {
+                    "id": "policy-id",
+                    "outputs": {
+                        "default": {
+                            "hosts": "https://localhost:9200"
+                        }
+                    },
+                    "streams": [
+                        {
+                            "id": "string",
+                            "type": "logs",
+                            "path": "/var/log/hello.log",
+                            "output": {
+                                "use_output": "default"
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    ],
+    "success": true
```

Comment:
@nchaulet Did we remove `success` from the fleet checkin response? I though all endpoints response was returning that field? But looking at your comments in https://github.com/elastic/kibana/pull/51968 this doens't appear to be the case?

Reply:
No it's not removed it's here, I just did not put it in the examples.